packages/amplify-category-function/src/events/prePushHandler.ts

```diff
@@ -11,8 +12,21 @@ import { ensureEnvironmentVariableValues } from '../provider-utils/awscloudforma
  * prePush Handler event for function category
  */
 export const prePushHandler = async (context: $TSContext): Promise<void> => {
-  await ensureEnvironmentVariableValues(context);
-  await ensureFunctionSecrets(context);
+  // if appId and envName can be resolved, proceed with checking env vars and secrets
+  const envName = stateManager.getCurrentEnvName() || context?.exeInfo?.inputParams?.amplify?.envName;
+  // get appId from amplify-meta or fallback to input params
+  const appId: string | undefined =
+    (stateManager.getMeta(undefined, { throwIfNotExist: false }) || {})?.providers?.awscloudformation?.AmplifyAppId ||
+    context?.exeInfo?.inputParams?.amplify?.appId;
+  if (envName && appId) {
+    await ensureEnvironmentVariableValues(context, appId);
+    await ensureFunctionSecrets(context);
+  } else {
+    printer.warn(
+      'Could not resolve either appId, environment name or both. Skipping environment check for function secrets and environment variables',
+    );
```

Comment:
What's a scenario for this case? Is something else taking care of these parameters later in the process?
If this is is specifically for `init --forcePush` then should we `if (init) {do what's in this PR} else {keep old logic}` ?

Reply:
Yes, this case should only ever hit on init --forcePush. I was trying to prevent the "prePushHandler" from having an if statement like "if (init)" but that could potentially work.

I wouldn't say there are things "taking care" of these parameters later but they are used later. There's certiantly an opportunity to pass in these parameters to some functions rather than have functions resolve these values internally but that was a thread I didn't want to pull on in this PR

Reply:
For the second case, If the appId or envName are missing on push, we'd also want to warn here. Although something else will probably fail if those things are undefined on push.
Is a comment giving a brief explanation of the `init --forcePush` scenario sufficient? I'm hesitant to change this condition because of all the whack-a-mole e2e failures that landed at this particular conditional

Reply:
Yep. sounds good.